pkg/util/file_indexer.go

```diff
@@ -83,6 +86,77 @@ func resolveFilePath(directory string) (string, error) {
 	return directory, nil
 }
 
+// gitignoreFilePath gives the filepath of the .gitignore file in the contex
+func gitignoreFilePath(directory string) (string, error) {
+	directoryFi, err := os.Stat(filepath.Join(directory))
+	if err != nil {
+		return "", err
+	}
+
+	switch mode := directoryFi.Mode(); {
+	case mode.IsDir():
+		// do directory stuff
+		return filepath.Join(directory, ".gitignore"), nil
+	case mode.IsRegular():
+		// do file stuff
+		// for binary files
+		return filepath.Join(filepath.Dir(directory), ".gitignore"), nil
+	}
+
+	return directory, nil
+}
+
+// Check .gitignore file exists or not
+// Add odo-file-index.json entry to .gitignore
+func checkGitignoreFile(directory string) {
+
+	ignoreFile, err := gitignoreFilePath(directory)
+	if err != nil {
+		log.Fatalf("file does not exist: %s", err)
+	}
+
+	_, err = os.Stat(ignoreFile)
+
+	if err == nil {
```

Comment:
using `err == nill` and then putting code within here is unorthodox. why exactly are we continuing after erroring here: https://github.com/openshift/odo/pull/2305/files#diff-ded22a7f93f540164727d9088a72def4R118 ?

Reply:
This indeed looks unusual. In Go, we generally:
1. have a check for `if err != nil` and do something (mostly do some cleanup if required and throw the error back to caller or exit the program)
2. then, in the same function that has `if err != nil` check, we have more code that's supposed to be executed if there was no error.

In pseodo-code it might look something like below:
```go
func myAwesomeFunction(abc string) error {
    ignoreFile, err := gitIgnoreFilePath(directory)

    if err != nil {
        // do what needs to be done if there's an error
    }
    // now write the code that's supposed to be executed if there's no error in the step above.
}

Reply:
Thanks @dharmit for pseudo-code.
I have updated all the changes. Can you please review once again and let me know if any modification is required further.